### ISSUE TITLE
Fingerprints should be generated after autoprefixing files

### ISSUE THREAD

**Issue by mirague** [kimroen/ember-cli-autoprefixer#38]
It would seem the generated fingerprint on `.css` files is **generated -before- the file's contents are autoprefixed**. See this diff, comparing the original `vendor.css` file (without autoprefixing) to the one post autoprefixing:
https://www.diffchecker.com/xk5ell28

Both of them have the following fingerprint:
```
<link rel="stylesheet" href="/assets/vendor-71a8f7f0ec9ee74bd7700d99a5f0b69d.css" integrity="sha256-Wfg3oOX5yWiadWxDW9FY1hyYG5JAGkQ4UUeaf7bTXmg= sha512-ykM/yDR0Yy/sRRHm2hNUUQtV/MgRLeX1RYc3wNVLRpIB47kbD5C4goWlbBK9077Y7hTM3XGcNtpBevekWsoHoQ==">
```

This is an issue when:
- the file is cached as `vendor-71a8f7f0ec9ee74bd7700d99a5f0b69d.css` (without autoprefixing)
- the contents change due to autoprefixing ([diff](https://www.diffchecker.com/xk5ell28)), the filename remains the same.
- the generated integrity hashes -do- change and suddenly the user's cached version conflicts and is not loaded: 

![image](https://user-images.githubusercontent.com/178117/30203317-0f340ada-9482-11e7-8532-8e136d89e362.png)

**Versions**
Ember: 2.15.1
broccoli-asset-rev: 
ember-cli-autoprefixer: 0.8.0

**Comment by mirague** [comment]
The underlying cause may be similar to that of https://github.com/rickharrison/broccoli-asset-rev/issues/118

**Comment by cbrevdev** [comment]
Oh nice find! ember-cli-autoprefixer is indeed being run after the fingerprinting stage.

    DEBUG="*:Fingerprint,*:AutoprefixerFilter,*:SRIHashAssets" ember build --prod

Shows:

    broccoli-filter:Fingerprint
    broccoli-persistent-filter:AutoprefixerFilter
    broccoli-caching-writer:SRIHashAssets

with mismatched asset fingerprint:

    MD5 (dist/assets/vendor-eaf3a09c2cb3eef1a98fd76408256234.css) = 25f52e8d963d263cc305c82b2ba54368


We can influence the order by modifying package.json to indicate that autoprefix must occur before broccoli-asset-rev:

    "ember-addon": {
        "configPath": "tests/dummy/config",
        "before": "broccoli-asset-rev"
    },

resulting in an adjusted build order:

    broccoli-persistent-filter:AutoprefixerFilter
    broccoli-filter:Fingerprint
    broccoli-caching-writer:SRIHashAssets

and corrected asset fingerprint:

    MD5 (dist/assets/vendor-25f52e8d963d263cc305c82b2ba54368.css) = 25f52e8d963d263cc305c82b2ba54368

I'll put together a PR.

**Comment by mirague** [comment]
Hey @cbrevdev, thanks for the quick reply and PR! I had no idea about the existence of the "before" mechanic - I'm happy you did!

**Comment by kimroen** [comment]
Thank you very much for this, both of you. This has been fixed in #39 and released as version 0.8.1 now! 🎉
